packages/framework/src/Model/Product/ProductOnCurrentDomainFacadeInterface.php

```diff
@@ -123,4 +123,9 @@ public function getProductsByCategory(Category $category, int $limit, int $offse
      * @return array
      */
     public function getProductsOnCurrentDomain(int $limit, int $offset, string $orderingModeId): array;
+
+    /**
+     * @return int
+     */
+    public function getProductsCountOnCurrentDomain(): int;
```

Comment:
Should it be called `getAllListableProductCount...`?

Reply:
I do not think so, as no method in this facade has it in name and most of them are getting data from listable methods in repository

Reply:
> no method in this facade has it in name and most of them are getting data from listable methods in repository

Yeah and it is really confusing. The interface/facade name is `ProductOnCurrentDomain...` not `ListableProductOnCurrentDomain...`.

And also you wrote the second confusing thing - `most of them are getting data from listable methods`. It seems wrong to me as I cannot be sure what products will I get unless I walk through the code.

Reply:
Well you should think of it from other side - OnCurrentDomainFacades are working with products, that can be displayed (are not hidden, have price name...)

Reply:
I know that somehow (deep in myself) but on other hand the name is really confusing here. But fixing it would take more than just renaming one method 🤔 .

Think of it from other side - you do not know what is behind this interface, you will use it, because it gives you products for given domain (fine - I know there are 50 products on any domain). Now comes a surprise  - the method will give me only 40 of them, because it is not obvious to mi it is limitting hidden products and some others as well. 😞 

I guess we cannot do anything to improve it here and now. But we should really thing about the naming and maybe consider renaming methods or the whole interface.